BLUESPAWN-client/src/util/configurations/RegistryKey.cpp

```diff
@@ -187,7 +190,15 @@ namespace Registry {
 	}
 
 	bool RegistryKey::ValueExists(const std::wstring& wsValueName) const {
-		return ERROR_SUCCESS == RegQueryValueExW(hkBackingKey, wsValueName.c_str(), nullptr, nullptr, nullptr, nullptr);
+		UNICODE_STRING RegistryKeyName{ wsValueName.length() * 2, wsValueName.length() * 2, const_cast<PWSTR>(wsValueName.c_str()) };
+
+		ULONG size{};
+		NTSTATUS status{ Linker::NtQueryValueKey(hkBackingKey, &RegistryKeyName, 0, nullptr, 0, &size) }; //First 0 is KeyValueBasicInformation
```

Comment:
You can check the value of status here to determine why the call failed. You don't need the second call